packages/cron/src/lib/metrics.js

```diff
@@ -0,0 +1,21 @@
+// eslint-disable-next-line no-unused-vars, @typescript-eslint/no-unused-vars
+import { Histogram, Registry } from 'prom-client'
+
+/**
+ * @exports
+ * @typedef {Histogram<"byteLength">} RetrievalDurationSecondsMetric
+ */
+
+/**
+ * @param {Registry} registry
+ * @returns {RetrievalDurationSecondsMetric}
+ */
+export function createRetrievalDurationSecondsMetric(registry) {
+  /** @type RetrievalDurationSecondsMetric} */
+  const metric = new Histogram({
+    name: 'retrieval_duration_seconds',
```

Comment:
We should also record the time it took to upload the data.

Reply:
I agree. Can we do it as fast-follow after merge or should I do before?

Reply:
Yes can be follow up